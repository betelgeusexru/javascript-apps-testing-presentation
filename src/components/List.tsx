import React, { FC } from "react";

type Props = {
  list?: string[];
};

export const List: FC<Props> = ({ list }) => {
  return (
    <div>
      {!Boolean(list?.length) && <h1>no items at all.</h1>}

      {Boolean(list?.length) && (
        <ul>
          <div>мы должны сломать тебя</div>
          {list?.map((item, index) => {
            return <li key={index}>{item}</li>;
          })}
        </ul>
      )}
    </div>
  );
};
