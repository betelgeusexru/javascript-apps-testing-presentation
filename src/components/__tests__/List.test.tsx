import renderer from "react-test-renderer";
import { List } from "../List";

describe("List component", () => {
  test("renders correct when empty list provided", () => {
    const tree = renderer.create(<List />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correct when list provided", () => {
    const tree = renderer
      .create(<List list={["apple", "banana", "tomato"]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
