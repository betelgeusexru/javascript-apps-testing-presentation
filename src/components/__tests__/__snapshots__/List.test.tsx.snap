// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders correct when empty list provided 1`] = `
<div>
  <h1>
    no items at all.
  </h1>
</div>
`;

exports[`List component renders correct when list provided 1`] = `
<div>
  <ul>
    <div>
      мы должны сломать тебя
    </div>
    <li>
      apple
    </li>
    <li>
      banana
    </li>
    <li>
      tomato
    </li>
  </ul>
</div>
`;
